specs/eventing/test-plan/error-signalling-conformance.md

```diff
@@ -0,0 +1,470 @@
+```
+
+---
+
+## [Pre] Create a Broker instance which reconciliation fails
+
+```
+kubectl create -f specs/eventing/test-plan/control-plane/error-signalling/broker-not-ready.yaml
+```
+
+## [Test] Broker failure is propagated to the Ready status condition
+
+The failure caused by the intentionally invalid spec is propagated to the `Ready` condition, which has a `status` of
+`False`, a severity of `""` (error), a non-empty `reason`, and a non-empty `message`:
+
+```
+kubectl get brokers/not-ready -o jsonpath='{.status.conditions[?(@.type=="Ready")].status}{";"}{.status.conditions[?(@.type=="Ready")].severity}{";"}{.status.conditions[?(@.type=="Ready")].reason}{";"}{.status.conditions[?(@.type=="Ready")].message}'
+```
+
+### [Output]
+
+```
+False;;ChannelTemplateFailed;Error on setting up the ChannelTemplate: Broker.Spec.Config configuration not supported, only [kind: ConfigMap, apiVersion: v1]
+```
+
+---
+
+## [Pre] Create a Trigger instance which reconciliation fails
+
+```
+kubectl create -f specs/eventing/test-plan/control-plane/error-signalling/trigger-not-ready.yaml
+```
+
+## [Test] Trigger failure is propagated to the Ready status condition
+
+The failure caused by the intentionally invalid spec is propagated to the `Ready` condition, which has a `status` of
+`False`, a severity of `""` (error), a non-empty `reason`, and a non-empty `message`:
+
+```
+kubectl get triggers/not-ready -o jsonpath='{.status.conditions[?(@.type=="Ready")].status}{";"}{.status.conditions[?(@.type=="Ready")].severity}{";"}{.status.conditions[?(@.type=="Ready")].reason}{";"}{.status.conditions[?(@.type=="Ready")].message}'
+```
+
+### [Output]
+
+```
+False;;BrokerDoesNotExist;Broker "does-not-exist" does not exist
+```
+
+---
+
+## [Pre] Create a Channel instance which reconciliation fails
+
+```
+kubectl create -f specs/eventing/test-plan/control-plane/error-signalling/channel-not-ready.yaml
+```
+
+## [Test] Channel failure is propagated to the Ready status condition
+
+The failure caused by the intentionally invalid spec is propagated to the `Ready` condition, which has a `status` of
+`Unknown` and a severity of `""` (error).
+
+```
+kubectl get channels/not-ready -o jsonpath='{.status.conditions[?(@.type=="Ready")].status}{";"}{.status.conditions[?(@.type=="Ready")].severity}'
+```
+
+### [Output]
+
+```
+Unknown;
+```
+
+---
+
+## [Pre] Create a Subscription instance which reconciliation fails
+
+```
+kubectl create -f specs/eventing/test-plan/control-plane/error-signalling/subscription-not-ready.yaml
+```
+
+## [Test] Subscription failure is propagated to the Ready status condition
+
+The failure caused by the intentionally invalid spec is propagated to the `Ready` condition, which has a `status` of
+`Unknown` and a severity of `""` (error).
+
+```
+kubectl get subscriptions/not-ready -o jsonpath='{.status.conditions[?(@.type=="Ready")].status}{";"}{.status.conditions[?(@.type=="Ready")].severity}'
+```
+
+### [Output]
+
+```
+Unknown;
```

Comment:
Same comment as the Channel regarding simulating a `False` condition instead of `Unknown`, so that the test can also assert that an actual error message and reason are propagated to the user.